juju/application.py

```diff
@@ -41,14 +41,14 @@ def status(self):
         """Get the application status, as set by the charm's leader.
 
         """
-        return self.data['status']['current']
+        return self.__getattr__('status')['current']
```

Comment:
`self.status` is the same as `self.data['status']` (because of the `__getattr__` that we have on `ModelEntity`. But that won't work for hyphenated names, like 'agent-status'.

My main complaints with this is that 1) it makes my eyes bleed 2) it's hard to understand why it's there unless you already know (difficult to maintain) and 3) we (or someone else) will forget to use this technique in the future.

I'd like to see us come up with a cleaner way to achieve the same result if possible.